hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/SecretKeyProtocolServerSideTranslatorPB.java

```diff
@@ -102,6 +104,16 @@ public SCMSecretKeyResponse processRequest(SCMSecretKeyRequest request)
             .setSecretKeysListResponseProto(getAllSecretKeys())
             .build();
 
+      case GetCheckAndRotate:
+        try {
+          return scmSecurityResponse
+              .setCheckAndRotateResponseProto(
+                  checkAndRotate(request.getCheckAndRotateRequest().getForce()))
+              .build();
+        } catch (TimeoutException e) {
+          LOG.error("Timeout occurred while executing checkAndRotate.", e);
```

Comment:
See the exception handling several lines down. I think this timeout error can be considered an internal error and get transferred to client side.